site/en/guide/data_performance.ipynb

```diff
@@ -0,0 +1,1142 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Execution time: 1.4374910180013103\n"
+     ]
+    }
+   ],
+   "source": [
+    "_batch_map_num_items = 25\n",
+    "def factorial(x):\n",
+    "    if x <= 1:\n",
+    "        return 1.0\n",
+    "    else:\n",
+    "        f = x * factorial(x - 1.0)\n",
+    "        return f\n",
+    "        \n",
+    "\n",
+    "iterate(\n",
+    "    ArtificialDataset(\n",
+    "        num_samples=_batch_map_num_items  # Generate more items\n",
+    "    ).map(  # Apply function one item at a time\n",
+    "        lambda x: factorial(8)\n",
+    "    ).batch(  # Batch and unbatch to compare with the vectorized version\n",
+    "        _batch_map_num_items, drop_remainder=True\n",
+    "    ).unbatch()\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![Scalar map](./scalar_map.svg)\n",
+    "\n",
+    "In this example, you can see that the mapped function is applied for each sample. While this function is very fast, it has some overhead that impact the time performance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Execution time: 1.453609274001792\n"
+     ]
+    }
+   ],
+   "source": [
+    "iterate(\n",
+    "    ArtificialDataset(\n",
+    "        num_samples=_batch_map_num_items\n",
+    "    ).batch(\n",
+    "        _batch_map_num_items, drop_remainder=True\n",
+    "    ).map(  # Apply function on a batch of items\n",
+    "        lambda x: tf.fill(x.shape, factorial(8))\n",
+    "    ).unbatch()\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "scrolled": false
+   },
+   "source": [
+    "![Vectorized map](./vectorized_map.svg)\n",
+    "\n",
+    "This time, the mapped function is called once and applies to a batch of sample. While the function takes more time to execute, the overhead appear only once, improving the overall time performance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Caching\n",
+    "\n",
+    "The `tf.data.Dataset.cache` transformation can cache a dataset, either in memory or on local storage.\n",
+    "This will prevent some operations (like data loading and preparation) to be executed for each epoch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Execution time: 0.35616327800016734\n"
+     ]
+    }
+   ],
+   "source": [
+    "iterate(\n",
+    "    ArtificialDataset()\n",
+    "    .map(  # Apply time consuming operations before cache\n",
+    "        lambda x: factorial(8)\n",
+    "    ).cache(\n",
+    "    ).map(  # Apply memory consuming operations after cache\n",
```

Comment:
since we already have an example of memory consuming vs time consuming operations below, I think we can simplify this by removing the second map.